main.c

```diff
@@ -79,18 +99,23 @@ void random_handler(struct page_table *pt, int page ) {
     }
 
     loadFrameIntoPage(pt, page, frame);
+    num_faults++;
 }
 
 void loadFrameIntoPage(struct page_table *pt, int page, int frame) {
-    printf("page fault: setting page %d to frame %d\n", page, frame);
+    // printf("page fault: setting page %d to frame %d\n", page, frame);
+    num_reads++;
     disk_read(disk, page, page_table_get_physmem(pt) + frame * BLOCK_SIZE);
-    page_table_set_entry(pt, page, frame, PROT_READ|PROT_WRITE);
+    page_table_set_entry(pt, page, frame, PROT_READ);
 }
 
 void evictPage(struct page_table *pt, int page) {
-    printf("eviction!\n");
-    int frame = pt->page_mapping[page];
-    disk_write(disk, page, page_table_get_physmem(pt) + frame * BLOCK_SIZE);
+    // Only write to disk if the page has been modified
+    if(pt->page_bits[page] == (PROT_READ|PROT_WRITE)) {
```

Comment:
I don't get this. Every time we load a page, we set it's bits to `PROT_READ|PROT_WRITE`, so every time we evict a page, the same thing happens.